charmcraft/commands/build.py

```diff
@@ -129,7 +129,8 @@ def __init__(self, args, config):
 
         self.buildpath = self.charmdir / BUILD_DIRNAME
         self.config = config
-        self.metadata = parse_metadata_yaml(self.charmdir)
+        #self.metadata = parse_metadata_yaml(self.charmdir)
+        self.metadata_name = "declarativepoc"
```

Comment:
A Charmcraft refactor will be needed here.

The problem is that in current code, the `metadata.yaml` is needed in several places to start/use the craft_parts machinery, and for declarative charms (at least if we stick to the "plugin" model), the `metadata.yaml` will be available only after running the declarative plugin (by craft_parts).